miasm2/analysis/depgraph.py

```diff
@@ -243,47 +246,52 @@ def pending(self):
         resolution"""
         return self._pending
 
-    def _get_modifiers_in_cache(self, depnode, force=False):
-        """Recursively find nodes in the path of @depnode which are modifiers.
-        Update the internal cache If @depnode is already managed (ie. in
-        @depnode_queued), abort
-        @depnode: DependencyNode
-        @force (optionnal): boolean, compute modifiers even if @depnode is a
-        modifier node.
-        """
-
-        # Base case
-        if depnode not in self._cache:
-            # Constant does not have any dependencies
-            return [depnode] if depnode.modifier else []
-
-        if depnode.modifier and not force:
-            return [depnode]
-
-        # Recursion
-        dependencies = self._cache[depnode]
-
-        out = set()
-        # Launch on each depnodes
-        parallels = []
-        for depnode in dependencies:
-            parallels.append(self._get_modifiers_in_cache(depnode))
-
-        if parallels:
-            for parallel in itertools.product(*(p for p in parallels if p)):
-                out.update(parallel)
-
-        return out
-
-    def clean_modifiers_in_cache(self):
+    def _get_modifiers_in_cache(self, nodes_heads):
+        """Find modifier nodes in cache starting from @nodes_heads.
+        Returns new cache"""
+        worklist = []
+        worklist_depnodes = list(nodes_heads)
+        cache = {}
+        forced_nodes = set(nodes_heads)
+        while worklist or worklist_depnodes:
```

Comment:
To reach the end of the loop, we first reach the end of `while worklist_depnodes` at line 274, so `if worklist_depnodes` is `False`. During the second loop, `worklist_depnodes` is never modified, so at the end of it (line 286), we both have `worklist_depnodes` and `worklist`evaluated to `False` in condition statements. So the loop will always occurs one time max, so this `while worklist or worklist_depnodes:` can be omitted.